newton/_src/geometry/support_function.py

```
@@ -0,0 +1,339 @@
+      - PLANE: half-width in x, half-length in y (lies in XY plane at z=0, normal along +Z)
+      - TRIANGLE: vertex B-A stored in scale, vertex C-A stored in auxillary
+    """
+
+    shape_type: int
+    scale: wp.vec3
+    auxillary: wp.vec3
+
+
+@wp.func
+def support_map(
+    geom: GenericShapeData, direction: wp.vec3, data_provider: SupportMapDataProvider
```

Ruff (0.13.3)

Unused function argument: `data_provider`

(ARG001)